source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/ObjectModelTest.cpp

```diff
@@ -1009,19 +1010,22 @@ namespace AdaptiveCardsSharedModelUnitTest
             auto body = card->GetBody();
             auto textBlock = std::static_pointer_cast<TextBlock>(body[0]);
             auto textBlockNoRequires = std::static_pointer_cast<TextBlock>(body[1]);
-            std::unordered_map<std::string, std::string> hostProvides { { "foobar", "2" } };
-            Assert::IsTrue(textBlock->MeetsRequirements(hostProvides));
-            Assert::IsTrue(textBlockNoRequires->MeetsRequirements(hostProvides));
-            hostProvides.clear();
-            Assert::IsFalse(textBlock->MeetsRequirements(hostProvides));
-            Assert::IsTrue(textBlockNoRequires->MeetsRequirements(hostProvides));
-            hostProvides.insert({ "foobar", "1.9.9.9" });
-            Assert::IsFalse(textBlock->MeetsRequirements(hostProvides));
-            Assert::IsTrue(textBlockNoRequires->MeetsRequirements(hostProvides));
-            hostProvides.clear();
-            hostProvides.insert({ "foobar", "99" });
-            Assert::IsTrue(textBlock->MeetsRequirements(hostProvides));
-            Assert::IsTrue(textBlockNoRequires->MeetsRequirements(hostProvides));
+
+            FeatureRegistration featureRegistration;
+            featureRegistration.AddFeature("foobar", "2");
+
+            Assert::IsTrue(textBlock->MeetsRequirements(featureRegistration));
+            Assert::IsTrue(textBlockNoRequires->MeetsRequirements(featureRegistration));
+            // hostProvides.clear();
+            // Assert::IsFalse(textBlock->MeetsRequirements(hostProvides));
+            // Assert::IsTrue(textBlockNoRequires->MeetsRequirements(hostProvides));
+            // hostProvides.insert({ "foobar", "1.9.9.9" });
+            // Assert::IsFalse(textBlock->MeetsRequirements(hostProvides));
+            // Assert::IsTrue(textBlockNoRequires->MeetsRequirements(hostProvides));
+            // hostProvides.clear();
+            // hostProvides.insert({ "foobar", "99" });
+            // Assert::IsTrue(textBlock->MeetsRequirements(hostProvides));
+            // Assert::IsTrue(textBlockNoRequires->MeetsRequirements(hostProvides));
```

Comment:
We should uncomment this or delete it

Reply:
derp. weekend development is apparently less rigorous :)

---
In reply to: [275445293](https://github.com/Microsoft/AdaptiveCards/pull/2684#discussion_r275445293) [](ancestors = 275445293)